base/deprecated.jl

```diff
@@ -788,6 +788,80 @@ function symperm{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, pinv::Vector{Ti})
         "Pkg.add(\"SuiteSparse\") to install SuiteSparse on Julia v0.5."))
 end
 
+# For deprecating vectorized functions in favor of compact broadcast syntax
+macro dep_vectorize_1arg(S, f)
+    :( @deprecate $(esc(f)){T<:$(esc(S))}(x::AbstractArray{T}) $(esc(f)).(x) )
+end
+macro dep_vectorize_2arg(S, f)
+    S = esc(S)
+    f = esc(f)
+    T1 = esc(:T1)
+    T2 = esc(:T2)
+    quote
+        @deprecate $f{T1<:$S,T2<:$S}(x::T1, y::AbstractArray{T2}) $f.(x,y)
+        @deprecate $f{T1<:$S,T2<:$S}(x::AbstractArray{T1}, y::T2) $f.(x,y)
+        @deprecate $f{T1<:$S,T2<:$S}(x::AbstractArray{T1}, y::AbstractArray{T2}) $f.(x,y)
+    end
+end
+
+# Deprecate @vectorize_1arg-vectorized functions from...
+for f in (
+        :sinpi, :cospi, :sinc, :cosc, # base/special/trig.jl
+        :log, :log1p, # base/special/log.jl
+        :gamma, :lfact, :digamma, :trigamma, :zeta, :eta,# base/special/gamma.jl
+        :erfcx, :erfi, :dawson, # base/special/erf.jl
+        :airyprime, :airyai, :airyaiprime, :airybi, :airybiprime, :airy, :airyx, :besselj0, :besselj1, :bessely0, :bessely1, # base/special/bessel.jl
+        :cbrt, :sinh, :cosh, :tanh, :atan, :asinh, :exp, :erf, :erfc, :exp2, :expm1, :exp10, :sin, :cos, :tan, :asin, :acos, :acosh, :atanh, #=:log,=# :log2, :log10, :lgamma, #=:log1p,=# :sqrt, # base/math.jl
+        :abs, :abs2, :angle, :isnan, :isinf, :isfinite, # base/floatfuncs.jl
+        :acos_fast, :acosh_fast, :angle_fast, :asin_fast, :asinh_fast, :atan_fast, :atanh_fast, :cbrt_fast, :cis_fast, :cos_fast, :cosh_fast, :exp10_fast, :exp2_fast, :exp_fast, :expm1_fast, :lgamma_fast, :log10_fast, :log1p_fast, :log2_fast, :log_fast, :sin_fast, :sinh_fast, :sqrt_fast, :tan_fast, :tanh_fast, # base/fastmath.jl
```

Comment:
I like that these are annotated with where they came from, but should spread over multiple lines when it gets this long